src/evaluate.jl

```diff
@@ -189,7 +189,10 @@ function evaluate(a::HomogeneousPolynomial{T}, vals::Array{S,1} ) where
     return suma
 end
 
-evaluate(a::HomogeneousPolynomial) = zero(a[1])
+function evaluate(a::HomogeneousPolynomial)
+    a.order == 0 && return a[1]
+    zero(a[1])
```

Comment:
`a[0]`?

Reply:
`a[1]` is correct, since `a` is a `HomogeneousPolynomial`